OPAL updates for Habanero

jk-ozlabs opened the pull request:
These changes implement a set of fixes for Habanero.

On the OPAL side, we update to skiboot-5.0.1, which means we no longer need the out-of-tree patch for IPOLL masking, and prevent a hang on XSCOM failures.

We bring in a few linux changes, mostly for preventing invalid error messages in non-FSP code

We also bring in a petitboot fix for the OS Boot sensor.

williamspatrick commented on buildroot:
```diff
@@ -1 +1 @@
-Subproject commit 9ac423f6f0d99b33e7bb7ad703cf83e80716da8a
+Subproject commit 80a2f83e60f3fd21a38e2bc9aac149f2c3a9ffdd
```
This commit is not valid.  @jk-ozlabs, did you forget to do a push?

williamspatrick commented:
Need to have latest patches from @ploetzma removed as well.

jk-ozlabs commented:
@williamspatrick yep, forgot the push. I've updated the buildroot tree. I've also rebased these changes to the new master-next, so the external patches are removed.

touchqvfb commented:
Hi williamspatrick,
         I  boot  op-build1.7, And there is an error message on IPL.
Can you give me some advise?thanks.
       What's the userdata1 mean?Can you give me some matirial of xscom? I don't the function and theory of xscom.
UserData1 HMER value (piberr in bits 21:23) : 0x00c0040000000000

In spec bits 21:23:
HMER0_XSCOM_STATUS: 000 = no error
001 = XSCOM blocked due to pending error
010 = chiplet offline
011 = partial good
100 = invalid address / address error
101 = clock error
110 = parity err
111 = time out

110.60816|================================================ 
110.64730|Error reported by xscom (0x0400) 
110.65466| XSCom access error 
110.65467| ModuleId 0x07 XSCOM_DO_OP 
110.65467| ReasonCode 0x0401 XSCOM_STATUS_ERR 
110.65467| UserData1 HMER value (piberr in bits 21:23) : 0x00c0040000000000 
110.65468| UserData2 XSCom address : 0x00000000006b0328 
110.65468|User Data Section 0, type UD 
110.65468| Subsection type 0x05 
110.65468| ComponentId errl (0x0100) 
110.65469|User Data Section 1, type UD 
110.65469| Subsection type 0x15 
110.65469| ComponentId hb-trace (0x3100) 
110.65469|User Data Section 2, type UD 
110.65470| Subsection type 0x04 
110.65470| ComponentId errl (0x0100) 
110.65470|User Data Section 3, type UD 
110.65470| Subsection type 0x06 
110.65471| ComponentId errl (0x0100) 
110.65471| CALLOUT 
110.65471| HW CALLOUT 
110.65471| Reporting CPU ID: 8 
110.65471| Called out entity: 
110.65472|User Data Section 4, type UD 
110.65472| Subsection type 0x06 
110.65472| ComponentId errl (0x0100) 
110.65472| CALLOUT 
110.65472| PROCEDURE ERROR 
110.65473| Procedure: 85 
110.65473|User Data Section 5, type UD 
110.65473| Subsection type 0x15 
110.65473| ComponentId hb-trace (0x3100) 
110.65474|User Data Section 6, type UD 
110.65474| Subsection type 0x15 
110.65474| ComponentId hb-trace (0x3100) 
110.65474|User Data Section 7, type UD 
110.65475| Subsection type 0x15 
110.65475| ComponentId hb-trace (0x3100) 
110.65475|User Data Section 8, type UD 
110.65475| Subsection type 0x03 
110.65476| ComponentId errl (0x0100) 
110.65476|User Data Section 9, type UD 
110.65476| Subsection type 0x01 
110.65476| ComponentId errl (0x0100) 
110.65476| STRING 
110.65477| Hostboot Build ID: 
110.65477|User Data Section 10, type UD 
110.65477| Subsection type 0x04 
110.65477| ComponentId errl (0x0100) 
110.65478|================================================

dcrowell77 commented:
@touchqvfb Please open a separate issue under the hostboot project for this discussion rather than using an arbitrary pull request.  Thanks.
